eng/common/scripts/Add-RetentionLease.ps1

```diff
@@ -18,29 +18,34 @@ param(
   [Parameter(Mandatory = $false)]
   [string]$OwnerId = "azure-sdk-pipeline-automation",
 
-  [Parameter(Mandatory = $false)]
+  # This script shouldn't need anything other than the $System.AccessToken from
+  # from the build pipeline. The retain-run.yml template doesn't run outside
+  # of the pipeline it's manipulating the retention leases for.
+  [Parameter(Mandatory = $true)]
   [string]$AccessToken = $env:DEVOPS_PAT
 )
 
 Set-StrictMode -Version 3
 
 . (Join-Path $PSScriptRoot common.ps1)
 
-$encodedAuthToken = Get-Base64EncodedToken $AccessToken
+$Base64EncodedToken=$null
+if (![string]::IsNullOrWhiteSpace($AccessToken)) {
```

Comment:
If this variable is required this check shouldn't be needed but doesn't hurt to have it still if you like.

Reply:
*sigh* Gah, I'll remove the if check.